src/main.c

```diff
@@ -1,8 +1,120 @@
+#include <iostream>
+#include <string>
+#include <ctime>
+#include <vector>
+#include <sstream>
+#include <iomanip>
+#include <string.h>
+#include <windows.h>
+
+/********************Functions start**********************/
+static const std::string base64_chars =
+        "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+        "abcdefghijklmnopqrstuvwxyz"
+        "0123456789+/";
+
+static const unsigned char b64_table[65] = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
+
+// Function to decode Base64 and return as std::string
+std::string base64_decode_to_string(const std::string& input) {
+const std::string base64_chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
+size_t input_len = input.length();
+std::vector<uint8_t> output;

```

Comment:
- You should replace `std::vector` with C-style arrays, declare static arrays or dynamically allocate memory using `malloc`.
- Access elements using `arr[index]` and resize dynamic arrays with `realloc`.
- Use loops or `memcpy` for copying, and maintain a separate variable to track the size of dynamic arrays since C arrays don't store size information.
- Use `sizeof` for static array size calculations.
- Don't forget to free dynamically allocated memory with `free`, and ensure proper bounds checking to avoid overflows.